src/main/java/org/jboss/tools/intellij/openshift/actions/OdoAction.java

```diff
@@ -36,12 +40,32 @@ protected OdoAction(Class... filters) {
     @Override
     public void actionPerformed(AnActionEvent anActionEvent, TreePath path, Object selected) {
         telemetrySender = new TelemetrySender(PREFIX_ACTION + getTelemetryActionName());
-        this.actionPerformed(anActionEvent, (Object) getElement(selected), getOdo(anActionEvent));
+        Odo odo = getOdo(anActionEvent);
+        if (odo == null) {
+          return;
+        }
+        this.actionPerformed(anActionEvent, (Object) getElement(selected), odo);
```

Comment:
at this point, we're sure that odo isn't null, so maybe add notNull annotation on the abstract might help for docs purpose.

Reply:
We're not sure that it isn't `null`. See:
```Java
       Odo odo = getOdo(anActionEvent);
        if (odo == null) {
          return;
        }
```
getOdo() can return `null` in many cases:

```
    private Odo getOdo(AnActionEvent anActionEvent) {
        try {
          Tree tree = getTree(anActionEvent);
          if (tree == null) {
            return null;
          }
          ApplicationsTreeStructure structure = (ApplicationsTreeStructure) tree.getClientProperty(Constants.STRUCTURE_PROPERTY);
          if (structure == null) {
            return null;
          }
          ApplicationsRootNode root = (ApplicationsRootNode) structure.getApplicationsRoot();
          if (root == null) {
            return null;
          }
          return root.getOdo().getNow(null);
        } catch(Exception e) {
          LOGGER.warn("Could not get odo: " + e.getMessage(), e);
          return null;
        }
    }
```

Reply:
maybe i'm wrong but 
```
        if (odo == null) {
          return;
        }
```
will break execution if `odo` is null, meaning that  on next line,

`this.actionPerformed(anActionEvent, (Object) getElement(selected), odo);`

odo isn't null for the rest of the execution ?